pkg/backingstore/backingstore.go

```diff
@@ -367,21 +373,51 @@ func createCommon(cmd *cobra.Command, args []string, storeType nbv1.StoreType, p
 
 // RunCreateAWSS3 runs a CLI command
 func RunCreateAWSS3(cmd *cobra.Command, args []string) {
+	log := util.Logger()
 	createCommon(cmd, args, nbv1.StoreTypeAWSS3, func(backStore *nbv1.BackingStore, secret *corev1.Secret) {
 		targetBucket := util.GetFlagStringOrPrompt(cmd, "target-bucket")
 		region, _ := cmd.Flags().GetString("region")
 		secretName, _ := cmd.Flags().GetString("secret-name")
+		mandatoryProperties := []string{"AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"}
 
 		if secretName == "" {
 			accessKey := util.GetFlagStringOrPromptPassword(cmd, "access-key")
 			secretKey := util.GetFlagStringOrPromptPassword(cmd, "secret-key")
 			secret.StringData["AWS_ACCESS_KEY_ID"] = accessKey
 			secret.StringData["AWS_SECRET_ACCESS_KEY"] = secretKey
-		} else {
-			mandatoryProperties := []string{"AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY"}
+			secret.Data = make(map[string][]byte)
+			secret.Data["AWS_ACCESS_KEY_ID"] = []byte(base64.StdEncoding.EncodeToString([]byte(accessKey)))
+			secret.Data["AWS_SECRET_ACCESS_KEY"] = []byte(base64.StdEncoding.EncodeToString([]byte(secretKey)))
+		} else {	
 			util.VerifyCredsInSecret(secretName, options.Namespace, mandatoryProperties)
 			secret.Name = secretName
+			secret = util.GetSecretFromSecretReference(&corev1.SecretReference{
+				Name:      secret.Name,
+				Namespace: secret.Namespace,
+			})
+			util.SecretResetStringDataFromData(secret)
+		}
+
+		suggestedSecret := util.CheckForIdenticalSecretsCreds(secret, mandatoryProperties)
+
+		if suggestedSecret != nil {
+
+			var decision string
+			for {
+				log.Printf("Found a Secret in the system with the same credintials (%s), do you want to use it for this Backingstore? y/n", suggestedSecret.Name)
+				fmt.Scanln(&decision)
```

Comment:
do we still need it? didn't we say in our discussion that we don't need it at all and we will update the secrets?
if your answer is still yes I feel that putting this code in every type of backingstore/namespacestore CLI creation command is unneeded and we can put this logic in the createCommon function